format/Message.fbs

```diff
@@ -21,10 +21,104 @@ include "Tensor.fbs";
 
 namespace org.apache.arrow.flatbuf;
 
+/// ------------------------------------------------------
+/// Buffer encoding schemes.
+/// -------------------------------------------------------
+
+/// Encoding for buffers representing integer as offsets from a reference value.
+/// This encoding uses less bits then the logical type indicates.
+/// It saves space when all values in the buffer can be represented with a
+/// small bit width (e.g. if all values in an int64 column are between -128
+/// and 127, then a bit-width of 8 can be be used) offset from the
+/// reference value.
+table FrameOfReferenceIntEncoding {
+  /// The value that all values in the buffer are relative to.
+  reference_value: long = 0;
+  /// The range of integers the set may contain. Unsigned integers up to 32 bit
```

Comment:
Why not 64-bit unsigned integers? That sounds like an unpleasant oddity.

Reply:
Also, I suppose these are little-endian integers?

Reply:
By the way, if this is purely an optimization and doesn't have any user-visible semantic value, perhaps the signed / unsigned distinction is unnecessary.

Reply:
I was thinking leaving the unsigned 64 off the table would make java implementation easier/possible.  I'm happy to include it.

Reply:
removed the uint64_t restriction.

Reply:
the endianness should be governed by the endianess of the schema, I've updated to reflect this.